### ISSUE TITLE
[error] configure: error: ssl library missing

### ISSUE THREAD

**Issue by asfo** [allinurl/goaccess#1129]
Hi, I'm trying to run GoAccess with HTTPS but I can't even compile it in a Docker container because it shows the error:

`configure: error: ssl library missing`

I already have the libssl-dev installed but when I try to compile my container it prints that error.

I added the `--with-openssl` in the configuration:
https://goaccess.io/man#configuration

If I remove it, it compiles correctly but I can't use this without SSL support.

PS: I'm using the 1.2 version

Thanks for your support.

**Comment by allinurl** [comment]
Did you follow [**this** ](https://github.com/allinurl/goaccess#docker )instructions or did you build your custom image? Those instructions should give you an image with ssl built-in.

**Comment by allinurl** [comment]
I'm not sure I follow what you are trying to do. Are you trying to run goaccess on its own container and output a real-time ssl-enabled HTML report? If I follow the [instructions](https://github.com/allinurl/goaccess#docker) I posted above, I can get that report though.

**Comment by asfo** [comment]
Is easy:

->I create an Nginx container using `FROM ubuntu` on the Dockerfile
->Install everything from Nginx normally
->Include GoAccess with `--with-openssl` flag

And doesn't works.

The instructions here:
```
$ wget http://tar.goaccess.io/goaccess-1.2.tar.gz
$ tar -xzvf goaccess-1.2.tar.gz
$ cd goaccess-1.2/
$ ./configure --enable-utf8 --enable-geoip=legacy
$ make
# make install

**Comment by allinurl** [comment]
I think I understand. You said that you tried to run it outside of the container and install it at the server-level, but didn't work. Can I ask you which version of Ubuntu are you running so I can try to reproduce this from my side? Thanks.

**Comment by asfo** [comment]
Here it is:

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04 LTS"
NAME="Ubuntu"
VERSION="18.04 LTS (Bionic Beaver)"

I apologize if this is my fault and I didn't follow the instructions correctly or something was missing and I really appreciate your help.

**Comment by allinurl** [comment]
Support for OpenSSL 1.1* has been added. Please build from [**development**](https://github.com/allinurl/goaccess#build-from-github-development) and let me know if that works for you.

**Comment by asfo** [comment]
Sorry for the delayed response, I'm going to try with that and comment the results.

**Comment by asfo** [comment]
I tried and this is the error now:

```
Cloning into 'goaccess'...
autoreconf: Entering directory `.'
autoreconf: running: autopoint --force
Can't exec "autopoint": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 345.
autoreconf: failed to run autopoint: No such file or directory
autoreconf: autopoint is needed because this package uses Gettext
```

We have "gettext" installed and I read something about these libraries:
- libgeoip-dev 
- ñibncursesw5-dev
- libcppunit-dev

So installed them and nothing...still that error.

**Comment by asfo** [comment]
Yes, I did it also (sorry I didn't included it in my last comment), also installed:
- autotools-dev 
- libltdl-dev 
- autopoint

**Comment by asfo** [comment]
I just reinstalled it and now is showing this:

`configure: error: *** Missing development libraries for ncursesw` 

So I'm going to try to install it and I will update in a moment.

**Comment by asfo** [comment]
```

**Comment by allinurl** [comment]
What error are you getting? I just spun up a vanilla ubuntu 18.04 and I'm able to build fine with `--with-openssl`.

**Comment by asfo** [comment]
We reinstall everything from scratch (our test server is using Ubuntu 18.04 and is on DigitalOcean) and same issue, I tried installing:

-- libgeoip-dev
- autopoint
- libncursesw5-dev

And then trying to run 
autoreconf -fiv and shows the error previously commented ([all-recursive] Error 1) 

Everything from development.

If I do the same with 1.2 without

**Comment by allinurl** [comment]
It sounds like there's some sort of package conflict. Are you able to test this out on a fresh/clean install of Ubuntu 18.04? The packages I installed were: make, gcc, autoconf, autopoint, libncursesw5-dev, libssl-dev, git.

**Comment by asfo** [comment]
Yes, I tried in a fresh/clean install of Ubuntu 18.04.

I installed make, gcc, gettext, autoconf, autopoint, libncurses, libgeoip-dev, and others.

Also, I'm using openssl 1.1, I tried with 1.0.2, 1.1.1 and nothing. But as said it works actually on console without the SSL, only when `--with-openssl` flag is enabled it doesn't works.

**Comment by allinurl** [comment]
Unfortunately can't replicate this from my side. I built it again and it worked well:

    ./configure --enable-utf8 --with-openssl

Machine:

    $ uname -a
    Linux user 4.15.0-22-generic #24-Ubuntu SMP Wed May 16 18:15:17 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

libssl:

```
$ apt-cache policy libssl-dev
libssl-dev:
  Installed: 1.1.0g-2ubuntu4
  Candidate: 1.1.0g-2ubuntu4
  Version table:
 *** 1.1.0g-2ubuntu4 500
        500 http://us.archive.ubuntu.com/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
```

Here's where libssl lives on the machine:

```
/usr/lib/x86_64-linux-gnu/libssl.a
/usr/lib/x86_64-linux-gnu/libssl.so
/usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
/usr/lib/x86_64-linux-gnu/libssl.so.1.1
/usr/lib/x86_64-linux-gnu/libssl3.so

```

**Comment by asfo** [comment]
Thanks for all your support @allinurl, at the end we cannot fix it from development/source or from 1.2 release, we used the GoAccess as a container using the volumes, so we include the volume on the access.log from the Nginx container, and then we included that access.log as a volume to the GoAccess container, so we can read it from there and like that, we included the report.html from it to the Nginx container, so both are reading using volumes and with that we fixed it and we can use it now with SSL support since we were able to connect to the GoAccess container in that way.

We cannot fix it at a server-level or included it inside the Nginx container but with those volumes, we did the trick.

I close this, but if someone have the same issue, I hope this can help.
